numpy/core/src/multiarray/datetime.c

```diff
@@ -2042,28 +2014,13 @@ add_minutes_to_datetimestruct(npy_datetimestruct *dts, int minutes)
 {
     int isleap;
 
-    /* MINUTES */
     dts->min += minutes;
-    while (dts->min < 0) {
-        dts->min += 60;
-        dts->hour--;
-    }
-    while (dts->min >= 60) {
-        dts->min -= 60;
-        dts->hour++;
-    }
 
-    /* HOURS */
-    while (dts->hour < 0) {
-        dts->hour += 24;
-        dts->day--;
-    }
-    while (dts->hour >= 24) {
-        dts->hour -= 24;
-        dts->day++;
-    }
+    /* propagate invalid minutes into hour and day changes */
+    dts->hour += extract_unit(&dts->min,  60);
+    dts->day  += extract_unit(&dts->hour, 24);
```

Comment:
Bug is here. This is passing an `int*` where a `uint64_t*` is expected. I'm really alarmed that doesn't result in a compilation error.

Reply:
Seems like we don't have `-Werror=incompatible-pointer-types` in effect

Reply:
indeed, it would be nice to run with `-Wall`, as per [NEP 2](https://numpy.org/neps/nep-0002-warnfix.html) which was deferred but the [1.3.0 release notes](https://numpy.org/devdocs/release/1.3.0-notes.html#improvements-to-build-warnings) mentioned it should be possible